source/framework/core/inc/TRestDataSet.h

```diff
@@ -101,6 +101,12 @@ class TRestDataSet : public TRestMetadata {
     /// The list of dataset files imported
     std::vector<std::string> fImportedFiles;  //<
 
+    /// A list of new columns added during the dataset generation
+    std::vector<std::string> fColumnNames;
+
+    /// The list of expressions correspoding to each fColumnNames element
+    std::vector<std::string> fColumnExpressions;
+
```

Comment:
I would change it to:
´´´
std::vector<std:pair<std::string, std::string >> fColumnNameExpressions;
´´´

Reply:
I am trying to use `std::pair`, but when I try to iterate:

```
    for( const auto &[cName,cExpression]: fColumnNameExpressions )
        finalList.emplace_back(cName);
        fDataSet.Define(cName, cExpression);
    }
```

it gives me error 

```
/home/jgalan/rest-framework/source/framework/core/src/TRestDataSet.cxx:355:25: error: ‘cName’ was not declared in this scope
         fDataSet.Define(cName, cExpression);
```

Reply:
The following is also not working for me:

```
    for( const auto &column: fColumnNameExpressions )
        finalList.emplace_back(column.first);
        fDataSet.Define(column.first, column.second);
    }
```

Reply:
This should work:
´´´
    for( const auto &[cName, cExpression] : fColumnNameExpressions ) {
        finalList.emplace_back(cName);
        fDataSet.Define(cName, cExpression);
    }
´´´
You forget the `}` ?

Reply:
ok